modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/logical/LogicalOrToUnionRule.java

```diff
@@ -155,6 +228,15 @@ private RelNode createUnionAll(RelOptCluster cluster, RelNode input, RexNode op1
             .build();
     }
 
+    /** */
+    private static boolean preMatch(IgniteLogicalTableScan scan) {
+        return scan.condition() != null &&
+            !F.isEmpty(scan.requiredColumns()) &&
```

Comment:
Why `requiredColumns` is a preMatch condition here?

Reply:
i just suppose we no need such a scans too, i`m wrong ?

Reply:
Why not? `requiredColumns` is just an optimization to filter out required columns on the source node. If you require all columns (for example in the query like `SELECT _key, _val WHERE a = 1 OR b = 1`) this set will be empty, but we still need to split such queries.